### PR TITLE
feat(ingestion): implement Pull-A-Part and upullitne TAP connectors

### PR DESCRIPTION
**Pull request by AugusDogus**
## Summary
- implement a new `pullapart` vehicle source using the shared Pull-A-Part / U-Pull-&-Pay public APIs
- implement a new `upullitne` vehicle source for the TAP Inventory Search System plugin used by `upullitne.com`
- add both connectors to the ingestion pipeline, reconcile flow, status reporting, saved-search filters, Algolia/search mapping, ingestion scripts, and user-facing source labels/copy
- harden live decoding for real provider edge cases encountered during runtime validation and PR review
- optimize the `upullitne` TAP connector to use direct per-store full inventory dumps instead of slow store→make→model fanout

## Review follow-up fixes
- fix `run-pipeline.ts` source finalization so `upullitne` source runs are correctly recognized and finalized on failure paths
- keep missing-state advancement/deletion gating tied to the **core** sources (`row52`, `pyp`, `autorecycler`) so transient failures in supplemental hole-filling sources (`pullapart`, `upullitne`) do not unnecessarily freeze missing-state advancement
- fix Pull-A-Part region normalization to pass a missing abbreviation instead of duplicating the state name
- switch TAP bootstrap to dynamic nonce extraction from the inventory page instead of relying on a static hardcoded nonce
- move TAP search response decoding onto the Effect error channel instead of synchronous decode-only flow
- convert TAP `success: false` HTTP-200 responses into provider failures
- fail the TAP connector when no concrete stores are returned after filtering out the `Any` placeholder
- make TAP product decoding tolerant of live optional fields (`yard_in_date`, `hol_year`, `hol_mfr_code`, `hol_mfr_name`, `hol_model`)
- update TAP cross-store dedupe accounting so `vehiclesProcessed` only counts newly seen VINs across the connector run
- clarify that TAP `pricesUrl` intentionally aliases the same public parts/pricing page as `partsUrl`

## TAP / upullitne reverse engineering
- fetched the live inventory page and plugin bundle to recover the actual AJAX contract
- confirmed the plugin bootstraps with:
  - `sif_ajax_url = https://upullitne.com/wp-admin/admin-ajax.php`
  - `sif_ajax_nonce` embedded in `sif_ajax_object` on the rendered inventory page
- confirmed option endpoints:
  - `sif_get_stores`
  - `sif_get_makes`
  - `sif_update_models`
- confirmed `sif_search_products` requires the exact serialized form field names from the rendered page:
  - `sif_form_field_store`
  - `sif_form_field_make`
  - `sif_form_field_model`
  - `sorting[key]`, `sorting[state]`, `sorting[type]`
  - `sif_verify_request`
- deeper finding: this install supports a true **full-store dump**
  - `sif_search_products` with just `sif_form_field_store=<store>` plus sorting fields returns the store’s full inventory
  - omitting make/model behaves the same as explicit `Any/Any`
  - per-make fanout was unnecessary for this site

## Testing
- `SKIP_ENV_VALIDATION=1 bun run typecheck`
- `bun test src/server/ingestion/reconcile.test.ts src/server/ingestion/pipeline-policy.test.ts src/lib/__tests__/saved-search-filters.test.ts src/lib/__tests__/algolia-alert-search.test.ts tests/ingestion/run-pipeline.smoke.test.ts`
- Pull-A-Part live connector check:
  - `bun -e 'import { Effect } from "effect"; import { streamPullapartInventory } from "./src/server/ingestion/pullapart-connector"; const result = await Effect.runPromise(streamPullapartInventory({ onBatch: () => Effect.succeed(undefined), pagesPerChunk: 1000 })); console.log(JSON.stringify(result, null, 2));'`
  - completed successfully with `count: 43658`, `pagesProcessed: 1365`, `fullyExhausted: true`, `errors: []`
- TAP live connector check with dynamic nonce extraction and tolerant decoding:
  - `bun -e 'import { Effect } from "effect"; import { streamTapInventory } from "./src/server/ingestion/tap-inventory-connector"; const result = await Effect.runPromise(streamTapInventory({ onBatch: () => Effect.succeed(undefined), pagesPerChunk: 1000 })); console.log(JSON.stringify(result, null, 2));'`
  - completed successfully with:
    - `count: 2764`
    - `pagesProcessed: 4`
    - `fullyExhausted: true`
    - `errors: []`

## Notes
- Pull-A-Part yard coordinates are derived from normalized ZIP centroids because the public service payloads do not expose lat/lng.
- `upullitne` is modeled as a site-specific TAP source key rather than a generic `tap` family source for now, matching the current evidence and issue guidance.
- Source precedence now treats Pull-A-Part and U Pull-It Nebraska as lower-priority hole-filling sources after Row52 and PYP, ahead of AutoRecycler when those higher-priority sources are healthy.
- Trigger ingestion logging now includes both `pullapartCount` and `upullitneCount` so scheduled runs reflect the full pipeline output.

<div><a href="https://cursor.com/agents/bc-0dcdab18-485b-4a28-8328-3eee5a9f42b0"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-0dcdab18-485b-4a28-8328-3eee5a9f42b0"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Summary by CodeRabbit

* **New Features**
  * Added Pull‑A‑Part and U Pull‑It Nebraska as searchable vehicle sources and included them across ingestion/status pipelines.
  * Search filters, saved searches, and sidebar source list now accept and display the new sources.
* **UI/Stats**
  * Site copy and SEO metadata updated; dashboard/stats now show five yard networks and per-source vehicle counts for the new sources.
* **Tests**
  * Test coverage expanded to include the new sources.

### CODE REVIEW

**Comment by AugusDogus** [comment]
@cursoragent please fix these issues

**Comment by AugusDogus** [comment]
@cursoragent Verify each finding against the current code and only fix it if needed.

Inline comments:
In `@src/server/ingestion/run-pipeline.ts`:
- Around line 1083-1089: The array passed into shouldAdvanceMissingState
currently includes supplemental hole-filling outcomes (pullapartResult,
upullitneResult) which can block missing-state advancement on their transient
failures; create a separate core outcomes array (e.g., coreSourceOutcomes)
containing only the core sources (row52Result, pypResult, autorecyclerResult)
and pass that to shouldAdvanceMissingState instead of sourceOutcomes, leaving
sourceOutcomes (or the original list) intact for other uses; apply the same
change for the second occurrence referenced (the block around lines 1125-1129)
so both calls use only coreSourceOutcomes when calling
shouldAdvanceMissingState.

In `@src/server/ingestion/tap-inventory-client.ts`:
- Around line 313-338: searchTapInventory currently treats HTTP-200 responses
with success: false as a successful result; update the pipeline so that after
decoding (decodeSearchResponse) you inspect the returned
TapInventorySearchResponse and, if response.success === false, convert it into
an Effect failure (e.g., Effect.fail(new Error(response.message || "TAP search
failed"))) so provider-side errors propagate; locate the place that calls
decodeSearchResponse in searchTapInventory (uses tapRequest and
decodeSearchResponse) and insert an Effect.flatMap/flatMapError step that checks
response.success and returns an Effect.fail when false, otherwise returns
Effect.succeed(response).

In `@src/server/ingestion/tap-inventory-connector.ts`:
- Around line 83-99: After fetching stores via fetchTapStores (and mapping
errors to TapInventoryProviderError), ensure you detect the case where, after
filtering out store.value === "Any" placeholders, there are zero concrete
stores: compute concreteStores = stores.filter(s => s.value !== "Any") and if
concreteStores.length === 0 then fail the source by returning/raising an Effect
failure using TapInventoryProviderError (e.g., with cursor: "stores" and an
explanatory cause/message) before logging or entering the for loop so the
connector surfaces a provider failure instead of reporting healthy.

File: src/server/ingestion/run-pipeline.ts [hunk 1]
```diff
@@ -781,21 +1056,36 @@ export const ingestionPipeline: Effect.Effect<
...
+      ...upullitneResult.errors,
     );
 
     const sourceOutcomes: SourceOutcome[] = [
       row52Result,
       pypResult,
       autorecyclerResult,
+      pullapartResult,
+      upullitneResult,
     ];
```
[the remark this answers is not in the document]
Comment (AugusDogus) [reply src/server/ingestion/run-pipeline.ts:1089]: @coderabbitai was this fixed?

[the remark this answers is not in the document]
Comment (AugusDogus) [reply src/server/ingestion/run-pipeline.ts:1089]: @coderabbiai then can you mark this thread/comment as resolved and approve the PR?

[the remark this answers is not in the document]
Comment (AugusDogus) [reply src/server/ingestion/run-pipeline.ts:1089]: @coderabbitai I was talking to you, that's why I tagged you. If this issue is fixed in the current code then can you mark this thread/comment as resolved and approve the PR?

**Comment by AugusDogus** [comment]
@coderabbitai approve (if it's ready)
